### PR TITLE
[NP-2058] - FixedSizeDAO Improvement

### PR DESCRIPTION
**Pull request by nanopay-arthur**
Ref: https://nanopay.atlassian.net/browse/NP-2058
#TimeSpent: 2h

### CODE REVIEW

File: src/foam/dao/FixedSizeDAO.js [hunk 1]
```diff
@@ -16,34 +16,40 @@ foam.CLASS({
...
   ],
 
   methods: [
     {
       name: 'put_',
       javaCode: `
+        Count count = new Count();
+        count = (Count) this.getDelegate().select(count);
+        if ( count.getValue() + 1 > getPurgeSize() + getPurgeSize() * getPurgePercent() / 100 ) {
+          this.getDelegate()
```
Comment (kgrgreer) [inline src/foam/dao/FixedSizeDAO.js:54]: You don't need to do a skip() or orderBy()? What is the predicate for?

File: src/foam/dao/FixedSizeDAO.js [hunk 2]
```diff
@@ -16,34 +16,40 @@ foam.CLASS({
...
+            .where(getPredicate())
+            .removeAll();
```
Comment (jlhughes) [inline src/foam/dao/FixedSizeDAO.js:58]: why did you drop orderby, skip?

File: src/foam/nanos/notification/email/services.jrl [hunk 3]
```diff
@@ -28,8 +28,7 @@ p({
       .setJournalType(foam.dao.JournalType.SINGLE_JOURNAL)
       .setJournalName("emailMessages")
       .setFixedSize(new foam.dao.FixedSizeDAO.Builder(x)
-        .setComparator(foam.mlang.MLang.DESC(foam.nanos.notification.email.EmailMessage.CREATED))
-        .setSize(10000)
+        .setPurgeSize(10000)
```
Comment (jlhughes) [inline src/foam/nanos/notification/email/services.jrl]: please re-add the comparator and skip and size.
and the purgepercent default of 10 is fine.

File: src/foam/dao/FixedSizeDAO.js [hunk 4]
```diff
@@ -16,34 +16,40 @@ foam.CLASS({
...
+        }
```
Comment (jlhughes) [inline src/foam/dao/FixedSizeDAO.js:59]: Do the put first, so the data is there, should the purge take time.

**Review by jlhughes — CHANGES_REQUESTED** [review]

**Pull request merged**